### PR TITLE
EZP-31651: Extracted search logic from adminUI

### PR DESCRIPTION
**Pull request by ViniTou**
| Question      | Answer
| ------------- | ---
| Tickets       |  https://jira.ez.no/browse/EZP-31651
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

## Desc
 - SearchController is the same as in adminUI, just used Symfony `FormFactory` instead of custom one.
- SearchView builder / provider introduced.
- Simplified front SA search forms.
image

## Use in adminUI:
https://github.com/ezsystems/ezplatform-admin-ui/pull/1400

## TODO
As a followup I would like to extract whole logic from Controller into ViewBuilder / Listeners.
#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,306 @@
+Copyright (C) 1999-2017 eZ Systems AS. All rights reserved.
```
Comment (lserwatka) [inline LICENSE:1]: Wrong license file. You need to use GPL one.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+| Question      | Answer
+| ------------- | ---
+| Tickets       |  [EZP-XXXXX](https://jira.ez.no/browse/EZP-XXXXX) <!-- URLs to JIRA issue(s) (or N/A) -->
+| Bug fix?      | yes/no
+| New feature?  | yes/no
+| BC breaks?    | yes/no
+| Tests pass?   | yes/no
+| Doc needed?   | yes/no
+| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)
```
Comment (mikadamczyk) [inline .github/pull_request_template.md:9]:
```suggestion
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-search/blob/master/LICENSE)
```

File: src/bundle/DependencyInjection/Configuration/Parser/Search.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                            ->integerNode('limit')
+                                ->info('Number of results per page')
+                            ->end()
+                        ->end()
+                    ->end()
+                ->end()
+            ->end();
+    }
+
+    public function mapConfig(array &$scopeSettings, $currentScope, ContextualizerInterface $contextualizer)
```
Comment (michal-myszka) [inline src/bundle/DependencyInjection/Configuration/Parser/Search.php:40]:
```suggestion
    public function mapConfig(array &$scopeSettings, $currentScope, ContextualizerInterface $contextualizer): void
```

File: src/bundle/Controller/SearchController.php [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+namespace Ibexa\Platform\Bundle\SearchBundle\Controller;
+
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\SearchService;
+use eZ\Publish\API\Repository\SectionService;
+use eZ\Publish\API\Repository\Values\Content\Language;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\Pagination\Pagerfanta\ContentSearchHitAdapter;
+use eZ\Publish\Core\QueryType\QueryType;
+use EzSystems\EzPlatformAdminUi\Form\Data\Search\SearchData;
```
Comment (mikadamczyk) [inline src/bundle/Controller/SearchController.php:16]: maybe we want to "copy" SearchData, SearchType, and other similar classes to this package and deprecate original classes in Admin UI. Or lets remove them now from Admin UI

File: src/bundle/Resources/views/themes/standard/search/form.html.twig [hunk 5]
```diff
@@ -0,0 +1,67 @@
+{% trans_default_domain 'search' %}
+
+{{ form_start(form, {'attr': {'class': 'ez-search-form'}}) }}
+<div class="input-group px-4">
+    {{ form_widget(form.query) }}
+
```
Comment (mikadamczyk) [inline src/bundle/Resources/views/themes/standard/search/form.html.twig:6]:
```suggestion
```

File: composer.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    }
+  },
+  "require": {
+    "php": "^7.3",
+    "symfony/dependency-injection": "^5.0",
+    "symfony/http-kernel": "^5.0",
+    "symfony/http-foundation": "^5.0",
+    "symfony/config": "^5.0",
+    "symfony/form": "^5.0",
+    "symfony/event-dispatcher": "^5.0"
```
Comment (mikadamczyk) [inline composer.json:25]: `ezplatform-kernel` and `pagerfanta` should be added

**Pull request merged**